src/main/java/graphql/nadel/engine/UnapplyEnvironment.java

```diff
@@ -0,0 +1,52 @@
+package graphql.nadel.engine;
+
+import graphql.execution.ExecutionStepInfo;
+import graphql.execution.nextgen.result.ExecutionResultNode;
+import graphql.language.Field;
+import graphql.schema.GraphQLSchema;
+import graphql.util.TraverserContext;
+import graphql.util.TreeTransformerUtil;
+
+import java.util.List;
+import java.util.Map;
+import java.util.function.Consumer;
+
+public class UnapplyEnvironment {
+
+
+    public UnapplyEnvironment(TraverserContext<ExecutionResultNode> context,
+                              ExecutionStepInfo parentExecutionStepInfo,
+                              boolean isHydrationTransformation,
+                              boolean batched,
+                              Map<String, String> typeRenameMappings,
+                              GraphQLSchema overallSchema,
+                              List<Field> notTransformedFields) {
+        this.context = context;
+        this.parentExecutionStepInfo = parentExecutionStepInfo;
+        this.isHydrationTransformation = isHydrationTransformation;
+        this.batched = batched;
+        this.typeRenameMappings = typeRenameMappings;
+        this.overallSchema = overallSchema;
+        this.notTransformedFields = notTransformedFields;
+    }
+
+    public TraverserContext<ExecutionResultNode> context;
+    public ExecutionStepInfo parentExecutionStepInfo;
+    public boolean isHydrationTransformation;
+    public boolean batched;
+    public Map<String, String> typeRenameMappings;
+    public GraphQLSchema overallSchema;
+    public List<Field> notTransformedFields;
+
```

Comment:
public getters please.  I know its painful.

Wats to address this

* allow a new dependency like Immutables or AutoValve
* use an IDEA plugin like BOB
* writee out own IDEA plugin